examples/onchain-signer/test/CommentBox.ts

```diff
@@ -33,55 +34,60 @@ describe('CommentBox', function () {
     console.log('    . gasless pubkey', wallet.address);
   });
 
-  it('Should comment', async function () {
-    this.timeout(10000);
+  async function commentGasless(comment: string, plain: boolean) {
+    const provider = ethers.provider;
+
+    const innercall = commentBox.interface.encodeFunctionData('comment', [
+      comment,
+    ]);
 
     const prevCommentCount = await commentBox.commentCount();
+    let tx: string;
+    if (plain) {
+      tx = await gasless.makeProxyTxPlain(
+        await commentBox.getAddress(),
+        innercall,
+      );
+    } else {
+      tx = await gasless.makeProxyTx(await commentBox.getAddress(), innercall);
+    }
 
-    const tx = await commentBox.comment('Hello, world!');
-    await tx.wait();
+    // TODO: https://github.com/oasisprotocol/sapphire-paratime/issues/179
+    const response = await provider.broadcastTransaction(tx);
```

Comment:
calldata encryption could be checked here (requires sapphire-paratime import)
```suggestion
    const response = await provider.broadcastTransaction(tx);
    expect(isCalldataEnveloped(response.data)).eq(!plain);
```

also is the  TODO still open?
I used `await wrapEthereumProvider(provider).broadcastTransaction(tx)` and didn't got an error.

Reply:
> I used `await wrapEthereumProvider(provider).broadcastTransaction(tx)` and didn't got an error.

https://github.com/oasisprotocol/sapphire-paratime/issues/179 is not relevant anymore. Fixed.

Reply:
> calldata encryption could be checked here (requires sapphire-paratime import)

Good idea!